shared/src/native-src/loader.cpp

```diff
@@ -644,46 +644,69 @@ namespace shared
-        DWORD functionNameLength = 0;
-        hr = metadataImport->GetMemberProps(functionToken, &functionParentToken, functionName, NameBuffSize, &functionNameLength,
+        DWORD functionNameSize = 0;
+        hr = metadataImport->GetMemberProps(functionToken, &functionParentToken, functionName, NameBuffSize, &functionNameSize,
                 nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr);
         if (FAILED(hr))
         {
             Error("Loader::HandleJitCachedFunctionSearchStarted: Call to GetMemberProps(..) returned a FAILED HResult: " + ToString(hr) + ".");
             return S_FALSE;
         }
 
+        if (SpecificMethodToInjectName != functionName)
+        {
+            if (ExtraVerboseLogging && _loaderOptions.LogDebugIsEnabled)
+            {
+                Debug("Loader::HandleJitCachedFunctionSearchStarted:"
+                      " (functionId=" +
+                      ToString(functionId) +
+                      ") "
+                      " resulted in disableNGenForFunction=False.");
+            }
+            return S_OK;
+        }
+
         WCHAR typeName[NameBuffSize]{};
-        DWORD typeNameLength = 0;
+        DWORD typeNameSize = 0;
         DWORD typeFlags;
-        hr = metadataImport->GetTypeDefProps(functionParentToken, typeName, NameBuffSize, &typeNameLength, &typeFlags, NULL);
+        hr = metadataImport->GetTypeDefProps(functionParentToken, typeName, NameBuffSize, &typeNameSize, &typeFlags, NULL);
         if (FAILED(hr))
         {
             Error("Loader::HandleJitCachedFunctionSearchStarted: Call to GetTypeDefProps(..) returned a FAILED HResult: " + ToString(hr) + ".");
             return S_FALSE;
         }
 
-        bool disableNGenForFunction = (0 == memcmp(functionName, SpecificMethodToInjectName, sizeof(WCHAR) * (std::min)(NameBuffSize, functionNameLength)))
-                                        && (0 == memcmp(typeName, SpecificTypeToInjectName, sizeof(WCHAR) * (std::min)(NameBuffSize, typeNameLength)));
-
-        if (disableNGenForFunction)
+        if (SpecificTypeToInjectName != typeName)
         {
-            _specificMethodToInjectFunctionId = functionId;
+            if (ExtraVerboseLogging && _loaderOptions.LogDebugIsEnabled)
+            {
+                Debug("Loader::HandleJitCachedFunctionSearchStarted:"
+                      " (functionId=" +
+                      ToString(functionId) +
+                      ","
+                      " functionMoniker=\"" +
+                      ToString(typeName) + "." + ToString(functionName) +
+                      "\")"
+                      " resulted in disableNGenForFunction=False.");
+            }
+            return S_OK;
         }
 
-        if ((ExtraVerboseLogging || disableNGenForFunction) && _loaderOptions.LogDebugIsEnabled)
+        _specificMethodToInjectFunctionId = functionId;
+
+        if (ExtraVerboseLogging && _loaderOptions.LogDebugIsEnabled)
```

Comment:
In the previous implementation, disableNGenForFunction was true so if ExtraVerboseLogging is false, the log was done
It means that ExtraVerboseLogging should be removed from the test:
if (_loaderOptions.LogDebugIsEnabled)
{
... log
}